[Buildkite] Test Android Staging on `ami-074cf521d1861d251`

ParaskP7 opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->

Related: [buildkite-ci#679](https://github.com/Automattic/buildkite-ci/pull/679)

AMI Name: `android-build-image-6.36.0v1.11-rc-2`

---

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR changes the Buildkite agent from `android` to `android-staging`. This change is not meant to be merged, but rather used to verify that the new AMI `ami-074cf521d1861d251` works as expected.

---

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

Check CI and verify everything is working as expected with the `android-staging` agent.

- [x] [JDK 17](https://github.com/woocommerce/woocommerce-android/pull/14276/commits/d2fc1211e6906cb15ed8e71c463b3c281b937eca) -> [Build](https://buildkite.com/automattic/woocommerce-android/builds/30332/steps/canvas) -> [Prototype Build Scan - Infrastructure](https://scans.gradle.com/s/cxkf5tlbb5sh2#infrastructure) -> `Amazon Corretto OpenJDK Runtime Environment 17.0.14+7-LTS`
- [x] [JDK 21](https://github.com/woocommerce/woocommerce-android/pull/14276/commits/ece285d8a7b87267faac0de3e7c0c9c533f511ec) + [Detekt](https://buildkite.com/automattic/woocommerce-android/builds/30335/steps/canvas?sid=0197d585-27c8-48fe-9563-486d4f72da20#0197d585-2842-4eb8-8716-ca46389375f6/216-239) -> [Build](https://buildkite.com/automattic/woocommerce-android/builds/30337/steps/canvas) -> [Prototype Build Scan - Infrastructure](https://scans.gradle.com/s/ydl6n43zcz2tu#infrastructure) -> `Amazon Corretto OpenJDK Runtime Environment 21.0.6+7-LTS`

---

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

ParaskP7 committed:
CI: Use android-staging agent to test ami-074cf521d1861d251

AMI Name: android-build-image-6.36.0v1.11-rc-2

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ParaskP7 committed:
Build: Update jdk to 21

ParaskP7 committed:
Build: Fix Inconsistent jvm-target compatibility on detekt rules module

CI Build Failure: Inconsistent JVM-target compatibility detected for
tasks 'compileJava' (21) and 'compileKotlin' (17).
